### PR TITLE
#97 lap times

### LINKED ISSUE #97: Performance Mode - Lap Time Button
### Description

Starts recording lap time when enter/return is pressed, and stops recording when it is pressed again. Stores and displays fastest lap time, last lap time, and current lap time.

### Acceptance Criteria

- [ ] starts recording time the first time enter/return is pressed
- [ ] stops recording the second time it is pressed
- [ ] stores fastest, last, and current times
- [ ] displays times on performance screen
- [ ] handle edge cases (no time has been recorded)

### Proposed Solution

- [ ] use QElapsedTimer
- [ ] create controller files that have Q_PROPERTYs that store times
- [ ] ask chatGPT/google

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by mattrwang**
## Changes

on performance screen, pressing enter starts timer, pressing enter again stops timer and stores info in lap time component

## Notes

currently not working, resetting all to 1

## To Do

_Any remaining things that need to get done_

- [ ] fix resetting to 1 issue

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #97

### CODE REVIEW

File: NERODevelopment/content/SpeedMode.qml [hunk 1]
```diff
@@ -18,6 +17,13 @@ Item {
     property int currentSpeed: speedController.currentSpeed
     property int currentDraw: speedController.current
 
+    focus: true
```
Comment (Peyton-McKee) [inline NERODevelopment/content/SpeedMode.qml]: U dont need this. This gets set in the navigation controller

File: NERODesign/content/TimerDisplay.qml [hunk 2]
```diff
@@ -29,7 +29,7 @@ Item {
             widthValue: timerDisplay.width
             heightValue: timerDisplay.height / 3.3
             label: "CURRENT RUN"
-            value: "0"
+            value: "2000"
```
Comment (Peyton-McKee) [inline NERODesign/content/TimerDisplay.qml:32]: put this back to 0

File: NERODevelopment/src/controllers/speedcontroller.h [hunk 3]
```diff
@@ -92,6 +81,8 @@ public slots:
   void setCurrentDischarge(float);
   void setMaxCurrentDischarge(float);
 
+  virtual void enterButtonPressed() override;
```
Comment (Peyton-McKee) [inline NERODevelopment/src/controllers/speedcontroller.h]: this is not virtual

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
The timer is way way too big, and it is not automatically updating when it should be

**Comment by jr1221** [comment]
good to merge?

**Pull request merged**